scripts/check_known_pulsars.py

```diff
@@ -30,6 +32,102 @@
     logger.warn("ATNF database could not be loaded on disk. This may lead to a connection failure")
     ATNF_LOC = None
 
+def search_for_cal_srclist(obsid, cal_id, all_cal_returns=False, all_srclist_returns=False):
+    """
+    Given an obsid, searches common locations for the rts folder as well as the sourcelist
+
+    Parameters:
+    -----------
+    obsid: int
+        The observation ID
+
+    Returns:
+    --------
+    cal_path: string
+        The location of the calibrator path
+    srclist: string
+        The pathname of the sourcelist
+    """
+    comp_config = config.load_config_file()
+    base_dir = comp_config['base_product_dir']
+    cal_dir = os.path.join(base_dir, str(obsid), "cal", str(cal_id))
+    cal_dirs=[]
+    srclists=[]
+    #search all subdirectories
+    for root, dirs, files in os.walk(cal_dir):
+        if "rts" in dirs:
+            cal_dirs.append(os.path.join(root, "rts"))
+        for f in files:
+            if f.endswith(".txt") and "srclist" in f:
+                srclists.append(os.path.join(root, f))
+
+    #handle multiple rts foldes with user input
+    if not all_cal_returns and len(cal_dirs) > 1:
+        valid = False
+        while not valid:
+            print("Multiple RTS files found. Please choose one")
+            for i, a_dir in enumerate(cal_dirs):
+                print("{0}: {1}".format(i+1, a_dir))
+            choice = int(input("Choose a number between 1 and {0}: ".format(len(cal_dirs))))
+            if choice >= 1 and choice <= len(cal_dirs):
+                valid=True
+                my_cal_dir = cal_dirs[choice-1]
+                print("Using RTS directory: {}".format(my_cal_dir))
+                cal_dirs = [my_cal_dir]
+
+            else:
+                print("Not a valid choice")
+
+    #handle multiple sourcelist files with user input
+    if not all_srclist_returns and len(srclists) > 1:
+        valid = False
+        print("Multiple sourcelist files found. Please choose one")
+        while not valid:
+            for i, a_file in enumerate(srclists):
+                print("{0}: {1}".format(i+1, a_file))
+            choice = int(input("Choose a number between 1 and {0}: ".format(len(srclists))))
+            if choice >= 1 and choice <= len(srclists):
+                valid=True
+                my_srclist = srclists[choice-1]
+                print("Using sourcelist directory: {}".format(my_srclist))
+                srclists = [my_srclist]
+            else:
+                print("## Not a valid choice! ##")
+
+    return cal_dirs, srclists
+
+def upload_cal_files(obsid, cal_id, cal_dir_to_tar=None, srclist=None):
+
+
+    if not cal_dir_to_tar:
+        cal_dir_to_tar = search_for_cal_srclist(obsid, cal_id, all_srclist_returns=True)[0][0]
+    if not srclist:
+        srclist = search_for_cal_srclist(obsid, cal_id, all_cal_returns=True)[1][0]
```

Comment:
Should only do this is there isn't already a calibration online (put it in the else at line 124)

Reply:
updated